source/elements/oneDAL/source/programming_model/end_to_end_example.rst

```diff
@@ -1,111 +1,86 @@
 ==================
 End-to-end example
 ==================
-
-Below you can find a typical workflow of using |dal_short_name| algorithm on GPU.
-The example is provided for Principal Component Analysis algorithm (PCA).
+Below you can find a typical workflow of using oneDAL algorithm on GPU. The
+example is provided for :txtref:`Principal Component Analysis algorithm (PCA)
+<alg_pca>`.
 
 The following steps depict how to:
 
-- Pass the data
-- Initialize the algorithm
-- Request statistics to be calculated (means, variances, eigenvalues)
-- Compute results
-- Get calculated eigenvalues and eigenvectors
+- Read the data from CSV file
+- Run the training and inference operations for PCA
+- Access intermediate results obtained at the training stage
 
-#. Include the following header file to enable the DPC++ interface for
-   |dal_short_name|:
+#. Include the following header. This make all oneDAL declarations available.
 
    .. code-block::
-      :substitutions:
 
-      #include "|daal_in_code|_sycl.h"
+      #include "oneapi/dal.hpp"
+
+      #include <cassert>
+      #include <iostream>
 
 #. Create a DPC++ queue with the desired device selector. In this case,
    GPU selector is used:
 
-   .. parsed-literal::
-
-      cl::sycl::queue queue { cl::sycl::gpu_selector() };
-
-#. Create an execution context from the DPC++ queue and set up as the
-   default for all algorithms. The execution context is the |dal_short_name|
-   concept that is intended for delivering queue and device information
-   to the algorithm kernel:
-
-   .. parsed-literal::
-
-      |dal_namespace|::services::Environment::getInstance()->setDefaultExecutionContext(
-            |dal_namespace|::services::SyclExecutionContext(queue) );
+   .. code-block:: cpp
 
-#. Create a DPC++ buffer from the data allocated on host:
+      const auto queue = sycl::queue{ sycl::gpu_selector{} };
 
-   .. parsed-literal::
+#. All oneDAL declarations are in the ``oneapi::dal`` namespace,
+   so import all declarations from the ``oneapi`` namespace to use ``dal``
+   instead of ``oneapi::dal`` for brevity:
```

Comment:
```suggestion
#. Since all oneDAL declarations are in the ``oneapi::dal`` namespace,
   import all declarations from the ``oneapi`` namespace to use ``dal``
   instead of ``oneapi::dal`` for brevity:
```